### PR TITLE
{AKS} Raise InvalidArgumentValueError for azure cni + pod_cidr without overlay

### PR DESCRIPTION
**Pull request by FumingZhang**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

Raise InvalidArgumentValueError for azure cni + pod_cidr without overlay. Sync changes in PR [#25113](https://github.com/Azure/azure-cli/pull/25113).

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 1]
```diff
@@ -45,14 +45,16 @@
 from knack.prompting import prompt_y_n
 
 from azext_aks_preview._consts import (
+    CONST_EBPF_DATAPLANE_CILIUM,
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:48]: Using consts in later comparison. Also did some sorting here.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 2]
```diff
@@ -400,12 +404,83 @@ def get_load_balancer_managed_outbound_ip_count(self) -> Union[int, None]:
...
+                enable_validation=False
+            )
+            if network_plugin:
+                if network_plugin == CONST_NETWORK_PLUGIN_AZURE and pod_cidr:
+                    if self.get_network_plugin_mode() != CONST_NETWORK_PLUGIN_MODE_OVERLAY:
+                        raise InvalidArgumentValueError(
+                            "Please specify network plugin mode `overlay` when using pod_cidr or "
+                            "use network plugin `kubenet`. For more information about Azure CNI "
+                            "Overlay please see https://aka.ms/aksoverlay"
+                        )
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:450]: Here is the difference compared to azure-cli/acs.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 3]
```diff
@@ -400,12 +404,83 @@ def get_load_balancer_managed_outbound_ip_count(self) -> Union[int, None]:
...
     def get_network_plugin_mode(self) -> Union[str, None]:
-        """Get the value of network_plugin_mode
+        """Get the value of network_plugin_mode.
+
+        Note: Currently this parameter does not support being updated.
 
         :return: str or None
         """
-        return self.raw_param.get('network_plugin_mode')
+        # read the original value passed by the command
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:471]: As the option does not support to be updated. The change has no practical effect.

File: src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py [hunk 4]
```diff
@@ -563,6 +563,71 @@ def test_get_disable_pod_security_policy(self):
...
+                }
+            ),
+            self.models,
+            DecoratorMode.CREATE,
+        )
+        # fail on network_plugin not specified
+        with self.assertRaises(RequiredArgumentMissingError):
+            ctx_2.get_network_plugin()
+
+        # custom
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py:599]: This test case is updated to catch the expected error.

File: src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py [hunk 5]
```diff
@@ -563,6 +563,71 @@ def test_get_disable_pod_security_policy(self):
...
+            ),
+            self.models,
+            DecoratorMode.CREATE,
+        )
+        # fail on network_plugin_mode not specified as overlay
+        with self.assertRaises(InvalidArgumentValueError):
+            ctx_3.get_network_plugin()
+
+        # custom
+        ctx_4 = AKSPreviewManagedClusterContext(
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py:617]: New test case to see validation passed.

File: src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py [hunk 6]
```diff
@@ -563,6 +563,71 @@ def test_get_disable_pod_security_policy(self):
         with self.assertRaises(MutuallyExclusiveArgumentError):
             ctx_2.get_disable_pod_security_policy()
 
+    def test_get_network_plugin(self):
+        # default
+        ctx_1 = AKSPreviewManagedClusterContext(
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py:568]: Copied these first 2 cases from azure-cli/acs.

**Comment by yonzhan** [comment]
AKS

**Review by tyler-lloyd — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**